alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/reporter/AttributeHealthReportsStore.java

```diff
@@ -0,0 +1,89 @@
+package eu.xenit.alfresco.healthprocessor.reporter;
+
+import eu.xenit.alfresco.healthprocessor.plugins.api.HealthProcessorPlugin;
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthReport;
+import eu.xenit.alfresco.healthprocessor.reporter.api.NodeHealthStatus;
+import eu.xenit.alfresco.healthprocessor.util.AttributeStore;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.alfresco.util.Pair;
+
+/**
+ * {@link HealthReportsStore} implementation that persists stats and failing reports via the {@link
+ * org.alfresco.service.cmr.attributes.AttributeService}.
+ * </p>
+ * Reports and stats are stored using the following key value combination:
+ * <ul>
+ *     <li>health-processor.reports.${random-uuid}=${Pair(PluginClass, NodeHealthReport)}</li>
+ *     <li>health-processor.report-stats.${PluginClass}=${HashMap(NodeHealthStatus, Count)}</li>
+ * </ul>
+ */
+@RequiredArgsConstructor
+public class AttributeHealthReportsStore implements HealthReportsStore {
+
+    private static final String ATTR_KEY_REPORTS = "reports";
+    private static final String ATTR_KEY_REPORT_STATS = "report-stats";
+
+    private final AttributeStore attributeStore;
+
+    @Override
+    public void storeUnhealthyReport(Class<? extends HealthProcessorPlugin> pluginClass, NodeHealthReport report) {
+        Pair<Class<? extends HealthProcessorPlugin>, NodeHealthReport> pluginClassWithReport =
+                new Pair<>(pluginClass, report);
+        attributeStore.setAttribute(pluginClassWithReport, ATTR_KEY_REPORTS, UUID.randomUUID().toString());
+    }
+
+    @Override
+    public void recordReportStats(Class<? extends HealthProcessorPlugin> pluginClass, Set<NodeHealthReport> reports) {
+        HashMap<NodeHealthStatus, Long> oldStats = attributeStore.getAttribute(ATTR_KEY_REPORT_STATS, pluginClass);
```

Comment:
use Map instead of HashMap

Reply:
This `oldStats` field is later in this method stored (`attributeStore.setAttribute(...)`), the expected value there is a `Serializable`. The `Map` interface doesn't extend `Serializable`...